### DISCUSSION TITLE
📢  Did you forget to update your Copilot extension and have lost access to Copilot?

### DISCUSSION THREAD

**Discussion by davecheney** [community/community#6637] [Copilot Conversations]
Greetings Pilots!

TL;DR we update the VSCode Copilot extension frequently to remove bugs and add features. It's important to keep your extension up to date because older clients cannot communicate with the Copilot servers.

If you've lost access to your Copilot invite; please do the following:

1. Update your copilot extension on all the machines you have it installed.
2. If you've lost access to your invite, please comment below and I'll fix you up.

Happy skies. Dave

**Comment by SOV37** [comment]
Root

**Comment by mrinal18** [comment]
copilot install in my vscode is not really working properly even though it's installed. is that a bug?

**Reply by davecheney** [reply]
Can you tell me more about the problem? Which version of copilot? Which version of vscode? What do you see in the Github Copilot output window?

**Reply by mrinal18** [reply]
VSCODE Version: 1.61.2 (Universal)
Github copilot version: v1.6.3450

Problem: 
initially when i used copilot with comments, it gave me suggestions based on those comments. i am trying in python itself and it seems like copilot isnt responding. for most cases it is not able to generate in copilot window as well even for a function as simple as adding the two numbers 

<img width="1117" alt="Screen Shot 2021-10-26 at 9 47 47 AM" src="https://user-images.githubusercontent.com/17683007/138892308-134d19ae-11b2-48e6-a35a-11f0b97447cb.png">

i will only be seeing this for long time. and still it has not synthesized.

Note: i am not using any proxy or firewall. 


Thanks.

**Reply by davecheney** [reply]
Thank you. This is an unrelated problem, please open a new discussion thread.

**Reply by mrinal18** [reply]
Thanks a lot, i will open an issues discussion. Glad to know that there's no issue with my access.

**Comment by gpips** [comment]
Not sure what the problem is but there are no auto suggestions and when i use ctrl+enter I get "Synthesizing 0/10 solutions" but nothing happens. Just go accepted to preview. Not sure what the issue.

**Reply by davecheney** [reply]
Can you tell me more about the problem? Which version of copilot? Which version of vscode? What do you see in the Github Copilot output window?

**Reply by gpips** [reply]
Thank you for the quick reply. It is Copilot v1.6.3450, vscode v1.61.2, and where do I find the output window. If it is what comes up on the right it just shows "Synthesizing 0/10 solutions"

**Reply by davecheney** [reply]
@gpips looks like you have access to copilot. I suggest starting a new thread to discuss completions.

**Reply by gpips** [reply]
Not sure what you mean. Is there something I need to do after being accepted that I have not done yet?

**Reply by davecheney** [reply]
@gpips it sounds like copilot is working. If you see output in the View -> Output window, then Copilot is working -- it may not have a lot to suggest depending on the language you are coding in, but that's probably best addressed in a new thread.

**Reply by gpips** [reply]
Sounds good. Thank you @davecheney

**Comment by oott123** [comment]
Not sure if I have lost my invite. I'm using VSCode 1.61.2 and GitHub Copilot v1.6.3450 and I can't access to copilot at all (It just don't shows up, all commands failed with something like command not found). Meanwhile my VSCode 1.62.0-insider which have GitHub Copilot installed exactly the same version as normal build, that can work.

I'm pretty sure that I can use Copilot few hours ago, and I have logged out and logged in my GitHub accounts, restarting VSCode and nothing happens at all. I can't see anything in GitHub Copilot Output panel.

**Reply by davecheney** [reply]
Can you please paste the output of View -> Output; select GitHub copilot in the bottom right

**Reply by oott123** [reply]
@davecheney There's nothing at all, literally.

**Reply by davecheney** [reply]
Thanks for confirming. Are you behind a firewall that might be blocking outgoing connections?

**Reply by oott123** [reply]
@davecheney I have to use http proxy to access the Internet but GitHub login works fine. I'm running workspace inside WSL but workspace outside WSL won't work either.

**Reply by davecheney** [reply]
I don’t think Copilot supports http proxies at the moment.

**Reply by myn** [reply]
> I don’t think Copilot supports http proxies at the moment.

I think you might be right.  I've disabled HTTP Proxy Strict SSL in vs.code settings but still receiving an error upon attempting to invoke the intellisense:

[INFO] [default] [2021-10-27T15:49:12.824Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[ERROR] [default] [2021-10-27T15:49:13.091Z] Request Error: self signed certificate in certificate chain
[ERROR] [ghostText] [2021-10-27T15:49:13.092Z] Error on ghost text request Error: self signed certificate in certificate chain
[INFO] [ghostText] [2021-10-27T15:49:13.092Z] Breaking, no choices

**Comment by madsh93** [comment]


**Reply by davecheney** [reply]
I've reconfirmed your invite. Please stay on top of copilot updates so you don't loose access in the future.

**Reply by madsh93** [reply]
@davecheney thanks!! That was quick

**Comment by divineniiquaye** [comment]
@davecheney Hello, I requested to join the beta program almost two months. But only got invite for issues beta program. Please I will be glad if you can confirm my invite to use Copilot.

**Reply by davecheney** [reply]
@divineniiquaye sorry, I’m not in a position to add people to the beta, I can only reactivate you if you were running an older version of the client. We’re adding people as we have capacity, thanks for your patience and understanding.

**Comment by fjlenatz** [comment]
@davecheney 
I'm getting the following error:
command 'github.copilot.generate' not found
And I'm not seeing the copilot icon. I'm on a MAC with vscode. Tried the following:

- Deleting the extension cache on ~/.vscode and on Application Support's CachedExtensions, signing off, reinstalling.
- Enabled the following configuration: "enable-proposed-api":["github.copilot"]

With the version 1.1.1909 the icon appearrs, but it doesn't suggests me anything (at least on Javascript)

Any ideas?

**Reply by davecheney** [reply]
@fjlenatz version 1.1 is hard coded to serves that no longer exist, that’s why you get no completions. Please see https://github.com/github/feedback/discussions/6785 for difficulties running the current version.

**Comment by TarunavBA** [comment]
Hey @davecheney I am not able to access copilot, I already got the email. When I install copilot extension in vscode, I am not able to see the sign in prompt or the copilot logo in vscode's footer. Can you pls reaccept me? (@TarunavBA is my username). I also have a doubt, should we install the extension the same day as we got the email that we are into the program, or can I install a few days later?

**Reply by TarunavBA** [reply]
@davecheney I have also updated the extension. I got the confirmation email yesterday, 3:47 AM in India. (Yes i'm in India)

**Reply by davecheney** [reply]
@TarunavBA please see https://github.com/github/feedback/discussions/6785

**Comment by TarunavBA** [comment]
Hey @davecheney can you please resend me the invitation for GitHub copilot? Please 🙏🙏🙏. I have recieved the conformation mail yesterday but it says that I don't have an invitation yet. Please resend me the invitation.

**Reply by TarunavBA** [reply]
I went to https://GitHub.com/GitHub/copilot-preview/invitations and saw. It said I don't have acces so can you send me the invitation for copilot and the https://GitHub.com/GitHub/copilot-preview/ repository? Please.

**Comment by TarunavBA** [comment]
@davecheney i only got link to this repository. Not that private one. As shown in the pic. 
![Screenshot_2021-10-29-14-05-06-093](https://user-images.githubusercontent.com/89698728/139404007-59b1343a-70f7-498b-9399-fa7145eef95f.jpeg)

**Reply by TarunavBA** [reply]
I only got link to this discussion forum not that private one.

**Reply by davecheney** [reply]
The link to the forum in the email is out of date, this is the correct forum.

I have confirmed you have been invited to the technical preview.

**Reply by TarunavBA** [reply]
@davecheney when will I get the email?

**Reply by davecheney** [reply]
looks like you already did, https://github.com/github/feedback/discussions/6637#discussioncomment-1557627

**Reply by TarunavBA** [reply]
That was the old one. I didn't recieve any new mail regarding copilot.

**Reply by davecheney** [reply]
There is only one email, you're in.

**Reply by TarunavBA** [reply]
Ok @davecheney I'll try again when my network is good and get back to you. Thanks 🙏

**Reply by matronator** [reply]
Hi, @davecheney. I also received just this email with link to the docs, but when I try Copilot in VSCode, it says to join the waiting list again (which if I click that link sends me to a page saying it's already enabled and to install the extension). Also, I found some links to some repos somewhere in here and I can't get into any of them (apparently not invited).

So basically - am I invited or not? Are there some additional steps I need to complete?

Thanks!

**Reply by davecheney** [reply]
@matronator your invite is active. If copilot is not working please see https://github.com/github/feedback/discussions/6810

**Comment by dorseytm** [comment]
Been trying to sign in for the past couple of days but keep getting an error stating my certificate has expired in Visual Studio Code.

**Reply by davecheney** [reply]
Please see https://github.com/github/feedback/discussions/6810

**Reply by dorseytm** [reply]
I checked #6810 and I'm not aware of any VPN or software that would be intercepting SSL connections if that's what you are asking.

**Comment by juliobarbagallo** [comment]
Hey, can you help me out? I got it in another machine, but the MB was damaged, I got a another pc, installed VSC and copilot, but it didn't even show in at the bottom.

**Reply by davecheney** [reply]
@juliobarbagallo close and reopen VSCode. You must accept the terms of use to use copilot.

**Reply by juliobarbagallo** [reply]
Thank you very much!

**Reply by RhianMcClelland** [reply]
@davecheney  like a few people i also am missing the  icon and am getting the  `command 'github.copilot.generate' not found`.
I was reading through a few of the comments and this response here is the only thing i can think of that i haven't checked. I don't recall seeing a terms of use.  I've done several uninstalls and extension cache deletes. Is there a way to force it back?

**Reply by davecheney** [reply]
@RhianMcClelland if you have not accepted the terms, you will be reprompted every time you reopen vscode.

**Reply by davecheney** [reply]
@RhianMcClelland please see https://github.com/github/feedback/discussions/6810

**Reply by RhianMcClelland** [reply]
@davecheney Perhaps I accepted the very first time hard to say, I certainly haven't seen it pop up again. Unlike #6810 i can't see the copilot icon. And my output is completely empty like there has never been any attempted communication. I suspect its probably something to do with corporate network but I've tried a few different configurations with/without the vpn's  with no luck. Anything else to try other then waiting for a build with a more helpful error message?

**Reply by davecheney** [reply]
@RhianMcClelland from our experience, if your network is firewalled off, copilot won't work.

**Comment by TarunavBA** [comment]
@davecheney please reconfirm my access I have lost access, copilot is not asking me to accept the GitHub copilot terms and conditions.

**Reply by davecheney** [reply]
See https://github.com/github/feedback/discussions/7044
